### ISSUE TITLE
Mandate the securityScheme in opanapi spec

### ISSUE THREAD

**Issue by sgomath** [stoplightio/spectral#2325]
I have the below security scheme defined in my openapi spec.
```
{
    "openapi": "3.0.1",
     "components": {
      "securitySchemes": {
        "bearerAuth": {
          "type": "http",
          "scheme": "bearer",
          "bearerFormat": "jwt"
        }
      }
    }
}
```

the rule that i have is as below
```
{
  "formats": [
    "oas3",
    "oas2"
  ],
 "extends": [
    ["spectral:oas","off"]
  ],
  "rules": {
  "oas-api-security-scheme-missing": {
    "description": "Requires the security scheme.",
    "message": "Must define a  security scheme",
    "recommended": true,
    "given": "$.components.securitySchemes.bearerAuth",
    "severity": "error",
    "then": {
          "field": "bearerFormat",
          "function": "truthy"
        }
    
   }
  }
}
```

Expected output : The error message should be thrown as the securityScheme doesn't exist in the above openapi spec. But the spectral lint doesn't throw any errors or warnings for this. Can you suggest me what is wrong in this?

**Comment by magicmatatjahu** [comment]
@sgomath Hi! But your OpenAPI spec has `components.securitySchemes.bearerAuth` object with defined `bearerFormat` field, so spec is valid against rule. Or I don't see something 🤔

**Comment by sgomath** [comment]
Hi [magicmatatjahu](https://github.com/magicmatatjahu), Thanks for the reply, Yes you are right. If i remove the components block from the above openapi spec and run the spectral lint with the above ruleset, expecting 'oas-api-security-scheme-missing' should be thrown. But it's not throwing any errors. So just wanted to understand what am i missing in the above rule.

**Comment by magicmatatjahu** [comment]
@sgomath Because then you don't have `$.components.securitySchemes.bearerAuth`. If given part of spec pointed by `given` doesn't exist, spectral skip validation for given rule. You should use `given: '$'` and then retrieve `components.securitySchemes.bearerAuth` value in validation function and check if it exist or not.

**Comment by sgomath** [comment]
Thanks [agicmatatjahu](https://github.com/magicmatatjahu), It worked, thanks again for your help.
